lib/api/unittest/CDataFrameAnalysisSpecificationTest.cc

```diff
@@ -101,10 +104,12 @@ void CDataFrameAnalysisSpecificationTest::testCreate() {
 
     LOG_DEBUG(<< "Valid input");
     {
-        LOG_TRACE(<< jsonSpec("1000", "20", "100000", "2", "custom_ml", {}, "outlier_detection"));
+        LOG_TRACE(<< jsonSpec("foo", "1000", "20", "100000", "2", "custom_ml",
+                              {}, "outlier_detection"));
         api::CDataFrameAnalysisSpecification spec{
-            runnerFactories(), jsonSpec("1000", "20", "100000", "2",
+            runnerFactories(), jsonSpec("foo", "1000", "20", "100000", "2",
                                         "custom_ml", {}, "outlier_detection")};
+        CPPUNIT_ASSERT_EQUAL(std::string("foo"), spec.jobId());
```

Comment:
nit: can we use `{}` initialisers. I'd like to stick to this in new code: a couple of times it's caught unintentional (and bad) type coercions for me.
```suggestion
        CPPUNIT_ASSERT_EQUAL(std::string{"foo"}, spec.jobId());
```